micro_pumas_v1.F90

```diff
@@ -3641,6 +3641,37 @@ subroutine micro_pumas_tend (                                            &
            nitend(i,k) = nitend(i,k)+ nnuccd(i,k)+ &
                 (nnucct(i,k)+tmpfrz+nnudep(i,k)+nsacwi(i,k)+nmultg(i,k)+dum2A(i,k))*lcldm(i,k)+(nsubi(i,k)-nprci(i,k)- &
                 nprai(i,k))*icldm(i,k)+(nnuccri(i,k)+nmultrg(i,k))*precip_frac(i,k)
+
+           ! ice number limiter
+           !================================================================
+           ! NorESM local ice number limiter. As in NorESM2 /CESM2 but included 
+           if ((naai(i,k) > 0._r8) .and. (t(i,k) < icenuct) .and. &
+                (relhum(i,k)*esl(i,k)/esi(i,k) > 1.05_r8)) then
+              nimax = naai(i,k)*icldm(i,k)*deltat
+           else
+              nimax = 0.0_r8
+           end if
+! RaFSIP GS/PG
+           ! Note: dum2A(i,k) should (hopefully) be re-inserted below after changes
+           !       in quantity tested for >0
+           !       in sum of terms for nimax calculation
+           dum2A(i,k) = 0.0_r8
+           if (rafsip_on) then
+              dum2A(i,k) = SIP_RATE(i,k)
+           end if
+! RaFSIP GS/PG
+           if (nnucct(i,k)+nnuccc(i,k)+nnudep(i,k)+dum2A(i,k) > 0._r8) then
+              nimax = nimax+(nnucct(i,k)+nnuccc(i,k)+nnudep(i,k)+dum2A(i,k))*lcldm(i,k)*deltat
+           end if
+           if (do_cldice .and. (nitend(i,k) > 0._r8) .and.                    &
```

Comment:
You have an extra check here. We already know `do_cldice` is `.true.` because of the logic on 3627:
```suggestion
           if ((nitend(i,k) > 0._r8) .and.                    &
```